com.unity.visualeffectgraph/Shaders/RenderPipeline/Universal/Templates/VFXParticleBasicCube.template

```diff
@@ -8,5 +8,6 @@
 		${VFXInclude("Shaders/ParticleHexahedron/PassDepth.template"),IS_OPAQUE_PARTICLE}
 		${VFXInclude("Shaders/ParticleHexahedron/PassBasicForward.template")}
 		${VFXInclude("Shaders/ParticleHexahedron/PassShadowCaster.template"),USE_CAST_SHADOWS_PASS}
+		${VFXIncludeRP("Templates/ParticleHexahedron/PassForward2D.template")}
```

Comment:
So we generate pass 2D always with URP ? 

Isnt it mutually exclusive with regular forwardPass (in that case it's just a matter a renaming the pass) in a project or can a project have both 2D and 3D renderer?

Also can we filter the 2D pass if 2D renderer is not used or is this unknown ?

Reply:
Is it done like this in unlit shader graph? (Always generate both passes)

Reply:
> So we generate pass 2D always with URP ?

Yes, we are always generating both, at some point, I would like to add a specific option to generate one, another or both but it would require to move SRP output in URP package (to allow serialization of specific data for URP).

> Also can we filter the 2D pass if 2D renderer is not used or is this unknown ?

For now, it's unknown, I've asked [here](https://unity.slack.com/archives/C89KFUUCT/p1612799052163400) but since we can switch the URP asset in runtime, checking the actual list of target isn't enough.

> Isnt it mutually exclusive with regular forwardPass (in that case it's just a matter a renaming the pass) in a project or can a project have both 2D and 3D renderer?

You can definitely have 2D & 3D renderer cohabiting, a camera can render using 2D while another, in the same scene, can render using 3D. Not sure if it's a use case but you can imagine render your 3D Scene and a minimap using 2D of the same scene.
By the way, in the project VisualEffect_URP, we are now mixing the classic 3D forward renderer and the 2D Renderer.

> Is it done like this in unlit shader graph? (Always generate both passes)

A shadergraph unlit is visible (by default, there is maybe a node settings somewhere which provide exclusion).
If you are taking the example of [Lit.shader](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Shaders/Lit.shader) both pass are declared :
- Universal2D https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Shaders/Lit.shader#L336
- UniversalForward https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Shaders/Lit.shader#L372 (!= UniversalForwardOnly since it supports deferred)

... but for `Unlit.shader`, the "main" pass doesn't declare any `LightMode` :
https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Shaders/Unlit.shader#L28

... Looking at the documentation : https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@10.2/manual/urp-shaders/urp-shaderlab-pass-tags.html
=> ```If you do not set the LightMode tag in a Pass, URP uses the SRPDefaultUnlit tag value for that Pass.```
I missed this detail, maybe we can use the "SRPDefaultUnlit" fallback for both passes, it could drastically reduce the amount of change for this PR but, on the other hand, it will prevent us to exclude a pass later.

See also integration of `SRPDefaultUnlit` in
- URP renderer : https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Runtime/Passes/DrawObjectsPass.cs#L47
- 2D renderer : https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs#L29

Reply:
We still have the possibility to use `SRPDefaultUnlit` instead of `UniversalForwardOnly`, it will  reduce the file change but it will prevent us to do any specific change to 2D or 3D passes in the future.